openapi.yaml

```diff
@@ -1,279 +1,678 @@
+            items:
+              type: string
+              enum: [services]
       responses:
         '200':
           description: Successful response
-          schema:
-            $ref: '#/definitions/PetsResult'
+          content:
+            'application/json':
+              schema:
+                $ref: '#/components/schemas/LocationSetResult'
         '400':
-          $ref: '#/responses/400'
+          $ref: '#/components/responses/400'
         '500':
-          $ref: '#/responses/500'
-    post:
-      summary: Add a new pet
-      operationId: postPet
+          $ref: '#/components/responses/500'
+  /locations/{locationId}:
+    get:
       tags:
-        - pets
+        - locations
+      description: 'Finds location by ID'
+      operationId: getLocationByID
       parameters:
-        - in: body
-          name: body
+        - name: locationId
+          in: path
+          description: 'ID of location to be searched by'
           required: true
           schema:
-            $ref: '#/definitions/PetPostBody'
+            type: string
       responses:
-        '201':
-          description: Pet successfully created
+        '200':
+          description: Successful response
+          content:
+            'application/json':
+              schema:
+                $ref: '#/components/schemas/LocationResult'
+        '404':
+          $ref: '#/components/responses/404'
+        '500':
+          $ref: '#/components/responses/500'
+  /locations/{locationId}/services:
+    get:
+      tags:
+        - locations
+        - services
+      description: 'Lists services related to a location'
+      operationId: getServices
+      parameters:
+        - name: locationId
+          in: path
+          description: 'ID of location'
+          required: true
           schema:
-            $ref: '#/definitions/PetResult'
-        '400':
-          $ref: '#/responses/400'
-        '409':
-          description: Conflict
+            type: string
+      responses:
+        '200':
+          description: Successful response
+          content:
+            'application/json':
+              schema:
+                $ref: '#/components/schemas/ServicesResultObjects'
+        '404':
+          $ref: "#/components/responses/404"
+        '500':
+          $ref: "#/components/responses/500"
+  /services:
+    get:
+      tags:
+       - services
+      summary: Get services
+      description: 'Gets all `Service` objects. Can be filtered by parameters.'
+      operationId: getServicesList
+      parameters:
+        - in: query
+          name: filter[query]
+          description: 'Query to search through buildings. Looks through the building name, abbreviation.'
+          required: false
           schema:
-            $ref: '#/definitions/ErrorResult'
+            type: string
+        - $ref: "#/components/parameters/isOpen"
+        - $ref: '#/components/parameters/pageNumber'
+        - $ref: '#/components/parameters/pageSize'
```

Comment:
It seems more practical to me that you'd want to get a location bundled with its services than service bundled with its location, so I'm not sure it would be worth it to implement. Thoughts @tsoliangwu0130?

Reply:
Good questions. Since we currently do have relationship in service objects in v1, I think some people may want to know related locations when querying services. I'm not 100% sure, but I imagine the logic should be pretty similar between these two cases so ideally it shouldn't be too hard. I'm open on this though, so maybe we can get feedback from others on this. @Ruefa @kalcicd thoughts?

Reply:
¿Por que no los dos? I can think of use cases for both scenarios.

Reply:
Sounds like the consensus is we should do it -- I'll get it added in.

Reply:
On that note, should /locations/:id have an include parameter? That could save the user from making two calls to /locations/:id and /locations/:id/services